ignite/metrics/precision.py

```diff
@@ -93,153 +179,139 @@ class Precision(_BasePrecisionRecall):
+            Precision: tensor([0.2000, 0.5000, 0.0000], dtype=torch.float64)
+            Micro Precision: 0.2222222222222222
+            Macro Precision: 0.2333333333333333
+            Weighted Precision: 0.175
+            Samples Precision: 0.2
 
-        In binary and multilabel cases, the elements of `y` and `y_pred` should have 0 or 1 values. Thresholding of
-        predictions can be done as below:
+        Thresholding of predictions can be done as below:
 
-        .. testcode:: 5
+        .. testcode:: 4
 
             def thresholded_output_transform(output):
                 y_pred, y = output
                 y_pred = torch.round(y_pred)
                 return y_pred, y
 
-            metric = Precision(average=False, output_transform=thresholded_output_transform)
+            metric = Precision(output_transform=thresholded_output_transform)
             metric.attach(default_evaluator, "precision")
-            y_true = torch.Tensor([1, 0, 1, 1, 0, 1])
+            y_true = torch.Tensor([1, 0, 1, 1, 0, 1]).long()
             y_pred = torch.Tensor([0.6, 0.2, 0.9, 0.4, 0.7, 0.65])
             state = default_evaluator.run([[y_pred, y_true]])
             print(state.metrics["precision"])
 
-        .. testoutput:: 5
-
-            0.75
-
-        In multilabel cases, average parameter should be True. However, if user would like to compute F1 metric, for
-        example, average parameter should be False. This can be done as shown below:
-
-        .. code-block:: python
-
-            precision = Precision(average=False)
-            recall = Recall(average=False)
-            F1 = precision * recall * 2 / (precision + recall + 1e-20)
-            F1 = MetricsLambda(lambda t: torch.mean(t).item(), F1)
-
-    .. warning::
+        .. testoutput:: 4
 
-        In multilabel cases, if average is False, current implementation stores all input data (output and target) in
-        as tensors before computing a metric. This can potentially lead to a memory error if the input data is larger
-        than available RAM.
+            [0.5, 0.75]
 
 
+    .. versionchanged:: 0.5.0
+            Some new options were added to `average` parameter.
     """
 
-    def __init__(
-        self,
-        output_transform: Callable = lambda x: x,
-        average: bool = False,
-        is_multilabel: bool = False,
-        device: Union[str, torch.device] = torch.device("cpu"),
-    ):
-        super(Precision, self).__init__(
-            output_transform=output_transform, average=average, is_multilabel=is_multilabel, device=device
-        )
-
     @reinit__is_reduced
     def update(self, output: Sequence[torch.Tensor]) -> None:
         self._check_shape(output)
         self._check_type(output)
         y_pred, y = output[0].detach(), output[1].detach()
 
         if self._type == "binary":
-            y_pred = y_pred.view(-1)
-            y = y.view(-1)
+
+            y = to_onehot(y.view(-1), num_classes=2)
```

Comment:
OK, I see, you pass binary case into multiclass (2-classes) case. I think we have to keep computed result for the single class (=1) instead of both. Our public API and usage should not change and this PR shouldn't be BC-breaking for this case.